File: Chapter06/PlaneActor.cpp
//
//  PlaneActor.cpp
//  Game-mac
//
//  Created by Michael Lehmann on 15.01.21.
//  Copyright © 2021 Sanjay Madhav. All rights reserved.
//

#include "PlaneActor.h"
#include "Game.h"
#include "Renderer.h"
#include "MeshComponent.h"

PlaneActor::PlaneActor(Game* game)
    :Actor(game)
{
    SetScale(10.0f);
    MeshComponent* mc = new MeshComponent(this);
    mc->SetMesh(GetGame()->GetRenderer()->GetMesh("Assets/Plane.gpmesh"));
}

File: Chapter06_Veldrid/PlaneActor.cs
namespace Chapter06
{
    public class PlaneActor : Actor
    {
        public PlaneActor(Game game) : base(game)
        {
            Scale = 10.0f;
            var mc = new MeshComponent(this) {Mesh = Game.Renderer.GetMesh("Assets/Plane.gpmesh")};
        }
    }
}